anago

```diff
@@ -1133,15 +1137,37 @@ update_github_release () {
   # post release data
   logecho "$release_verb the $RELEASE_VERSION_PRIME release on github..."
   local changelog_url="$K8S_GITHUB_URL/blob/master/$CHANGELOG_FILE"
-  release_id=$($GHCURL $K8S_GITHUB_API/releases$id_suffix --data \
-   '{
-    "tag_name": "'$RELEASE_VERSION_PRIME'",
-    "target_commitish": "'$RELEASE_BRANCH'",
-    "name": "'$RELEASE_VERSION_PRIME'",
-    "body": "See [kubernetes-announce@](https://groups.google.com/forum/#!forum/kubernetes-announce) and ['$CHANGELOG_FILE']('$changelog_url'#'${RELEASE_VERSION_PRIME//\./}') for details.\n\nSHA256 for `kubernetes.tar.gz`: `'${sha256_hash}'`\n\nSHA512 for `kubernetes.tar.gz`: `'${sha512_hash}'`\n\nAdditional binary downloads are linked in the ['$CHANGELOG_FILE']('$changelog_url'#downloads-for-'${RELEASE_VERSION_PRIME//\./}').",
-    "draft": '$draft',
-    "prerelease": '$prerelease'
-    }' |jq -r '.id')
+  query_tmpl='{
+    "tag_name" : $tag,
+    "target_commitish": $target,
+    "name": $name,
+    "body": $body,
+    "draft": $draft,
+    "prerelease": $prerelease
+  }'
+  body_fmt='See [kubernetes-announce@](https://groups.google.com/forum/#!forum/kubernetes-announce) and [%s](%s) for details.
+  
+    SHA256 for `kubernetes.tar.gz`: `%s` 
+    SHA512 for `kubernetes.tar.gz`: `%s`
+  
+    Additional binary downloads are linked in the [%s](%s).'
+  body="$(
+    printf "$body_fmt" \
+      "$CHANGELOG_FILE" "${changelog_url}#${RELEASE_VERSION_PRIME//\./}" \
+      "$sha256_hash" "$sha512_hash" \
+      "$CHANGELOG_FILE" "${changelog_url}#downloads-for-${RELEASE_VERSION_PRIME//\./}"
+  )"
+  query_body="$( jq \
+    --arg tag "$RELEASE_VERSION_PRIME" \
+    --arg target "$RELEASE_BRANCH" \
+    --arg name "$RELEASE_VERSION_PRIME" \
+    --arg body "$body" \
+    --arg draft "$draft" \
+    --arg prerelease "$prerelease" \
```

Comment:
`draft` & `prerelease` [should be booleans](https://developer.github.com/v3/repos/releases/#create-a-release), therefore we should:
- use `--argjson` for those
- maybe (didn't trace back, where this variables come from) have a function `common::validate_bool` or such to validate if the strings we use here are valid JSON bools?

Reply:
Looks like we set both of these variables explicitly in the preceding lines so I believe they should be safe here

Reply:
Agreed to using `--argjson` for sure though!